src/main/java/seedu/address/model/person/Person.java

```diff
@@ -128,4 +128,8 @@ public Role getRole() {
     public int compareNamesIgnorecase(Person otherPerson) {
         return name.compareToIgnoreCase(otherPerson.name);
     }
+
+    public boolean hasPhoneNumber(String phoneNumber) {
+        return this.phone.value.equals(phoneNumber);
```

Comment:
This might be violating LOD since it is calling the method of a parameter of phone. A non-violation would be something like `this.phone.hasPhoneNumber(phoneNumber)`? 😆